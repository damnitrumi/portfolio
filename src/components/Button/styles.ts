import styled, { css } from "styled-components";

export const Wrapper = styled.button`
    ${() => css`
      border: none;
      background-color: transparent;
      cursor: pointer;
    `}
`;
