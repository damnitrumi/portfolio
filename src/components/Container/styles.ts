import styled, { css } from "styled-components";

export const Wrapper = styled.div`
    ${({ theme }) => css`
        background-color: ${theme.colors.primaryColor};
        padding: ${theme.paddings.containerDefault};
        height: 100vh;

        @media ${theme.media.ltaMedium}{
            padding: ${theme.paddings.containerLtaMedium};
        }
    `}
`;
