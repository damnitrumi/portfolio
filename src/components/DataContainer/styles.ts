import styled, { css } from "styled-components";

export const Wrapper = styled.div`
    ${() => css`
        max-width: 1100px;
        margin: 0 auto;
        height: 100%;
        padding: 200px 0;
    `}
`;
