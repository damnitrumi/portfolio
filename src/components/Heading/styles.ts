import styled, { css } from "styled-components";
import { HeadingProps } from ".";

const pickSize = (size: string) => {
  if (size) {
    return css`
      font-size: ${size};
    `;
  }
};

export const Title = styled.h1<HeadingProps>`
${({ size, as }) => css`
  color: white;
  ${pickSize(size)}
  font-weight: ${as == "h1" ? 500 : 400};
`}
`;
