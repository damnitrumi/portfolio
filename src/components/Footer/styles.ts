import styled, { css } from "styled-components";
import { Wrapper as Container } from "../Container/styles";

export const Wrapper = styled.div`
  ${() => css`
  `}
`;

export const FooterContainer = styled(Container)`
    background-color: transparent;
    height: auto;
    position: fixed;
    width: 100%;
    bottom: 0;
`;

export const LinksContainer = styled.div``;
