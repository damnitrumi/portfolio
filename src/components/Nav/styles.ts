import styled, { css } from "styled-components";
import { Wrapper as Container } from "../Container/styles";

export const Wrapper = styled.nav`
    ${() => css`
        display: flex;
        align-items: center;
        justify-content: space-between;
    `}
`;

export const imgContainer = styled.a`
    ${() => css`
        max-width: 150px;
    `}
`;

export const NavContainer = styled(Container)`
    background-color: transparent;
    height: auto;
    position: fixed;
    width: 100%;
    z-index: 1;

    @media ${({ theme }) => theme.media.ltaMedium}{
      background-color: ${({ theme }) => theme.colors.mediaGray};
    }
`;

export const bgMenu = styled.div`
    position: fixed;
    height: 100vh;
    width: 100%;
    top: 0;
    left: 0;
    background-color: rgba(0, 0, 0, 0.3);
    /* pointer-events: none; */
`;
