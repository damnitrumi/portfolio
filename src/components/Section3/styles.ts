import styled, { css } from "styled-components";
import { Wrapper as DataContainer } from "../DataContainer/styles";
import { Wrapper as Container } from "../Container/styles";

export const Wrapper = styled.section`
    ${({ theme }) => css`
      ${DataContainer}{
        flex-direction: row;
        justify-content: space-between;
        align-items: center;

        @media ${theme.media.ltaMedium}{
          flex-direction: column;
          gap: 20px
        }
      }

      ${Container}{
        @media ${theme.media.ltaMedium}{
          height: auto;
        }
      }
    `}
`;
export const Education = styled.div`
    ${({ theme }) => css`
      background-color: rgba(255, 255, 255, 0.2);
      flex: 0 1 500px;
      padding: 50px;
      transition: all 300ms ease-in-out;
      border-radius: 5px ;

      @media ${theme.media.ltaMedium}{
        flex: 0 1 auto;
      }

      &:hover{
        background-color: #c32865;
      }
    `}
`;

export const EducationData = styled.div`
    ${({ theme }) => css`
      padding: 20px 0;

      & + &{
        border-top: 1px solid ${theme.colors.white};
      }
    `}
`;

export const Experience = styled.div`
    ${({ theme }) => css`
    background-color: rgba(255, 255, 255, 0.2);
    flex: 0 1 500px;
    padding: 50px;
    transition: all 300ms ease-in-out;
    border-radius: 5px ;

    @media ${theme.media.ltaMedium}{
        flex: 0 1 auto;
      }

    &:hover{
        background-color: #c32865;
      }

    `}
`;

export const ExperienceData = styled.div`
    ${({ theme }) => css`
    padding: 20px 0;
    & + &{
        border-top: 1px solid ${theme.colors.white};
      }
    `}
`;
