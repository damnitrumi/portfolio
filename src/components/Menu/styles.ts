import styled, { css } from "styled-components";
import { MenuBehaviorProps } from ".";
import { LinksContainer as IconsContainerMock } from "../Footer/styles";
import { StyledIconBase } from "@styled-icons/styled-icon";

export const Wrapper = styled.div<MenuBehaviorProps>`
    ${({ theme, menuVisible }) => css`
    position: fixed;
    width: 100%;
    max-width: 500px;
    top: 0;
    right: -100%;
    transition: all .8s ease-in-out;

    @media ${theme.media.ltaMedium} {
      max-width: 100%;
    };

    ${
      menuVisible &&
      css`
        right: 0;
    `
    }
    `}
`;

export const ButtonContainer = styled.div`
  display: flex;
  justify-content: flex-end;
`;
export const LinksContainer = styled.div`
  padding: 30px 0;

  a{
    padding: 20px 0;
    color: ${({ theme }) => theme.colors.menuGray};
    transition: all 200ms ease-in;
  }

  a:hover{
    color: ${({ theme }) => theme.colors.white};
  }
`;

export const IconsContainer = styled(IconsContainerMock)`
  ${StyledIconBase}{
    color: ${({ theme }) => theme.colors.menuGray};
    &:hover{
      color: ${({ theme }) => theme.colors.hoverGreen};
    }
  }
`;
