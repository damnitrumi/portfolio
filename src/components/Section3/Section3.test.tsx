import { screen } from "@testing-library/react";
import { renderTheme } from "../../styles/render-theme";
import { Section3 } from ".";

describe("<Section3 />", () => {
  it("should render Section3", () => {
    renderTheme(<Section3 />);

    expect(
      screen.getByRole("heading", {
        name: "Unicesumar, Maringá-PR / Out/2017 - Ago/2020",
      }),
    ).toBeInTheDocument();
    expect(
      screen.getByText(
        "Curso Superior de Tecnologia (CST) em Análise de Sistemas de Computação",
      ),
    ).toBeInTheDocument();

    expect(
      screen.getByRole("heading", {
        name: "Vip Sistemas, Maringá-PR / Jul/2018 - Dez/2019",
      }),
    ).toBeInTheDocument();
    expect(
      screen.getByText(
        "Realizava a função de suporte de sistemas de TI, onde trabalhava com a manutenção de servidores (Locais e em nuvem) e banco de dados.",
      ),
    ).toBeInTheDocument();
  });
  it("should render", () => {
    const { container } = renderTheme(<Section3 />);

    expect(container).toMatchSnapshot();
  });
});
