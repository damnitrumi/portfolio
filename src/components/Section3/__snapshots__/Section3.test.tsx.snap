// Vitest Snapshot v1

exports[`<Section3 /> > should render 1`] = `
.c2 {
  background-color: #1b1b1b;
  padding: 30px 60px;
  height: 100vh;
}

.c4 {
  max-width: 1100px;
  margin: 0 auto;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c6 {
  color: #FFFFFF;
  font-size: 4.7rem;
  font-weight: 400;
}

.c9 {
  color: #FFFFFF;
  font-weight: 400;
}

.c10 {
  color: #FFFFFF;
  font-size: 1.6rem;
  font-weight: 400;
}

.c11 {
  color: #FFFFFF;
  font-size: 1.4rem;
  line-height: 30px;
}

.c0 .c3 {
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 20px;
}

.c5 {
  background-color: rgba(255,255,255,0.2);
  -webkit-flex: 0 1 500px;
  -ms-flex: 0 1 500px;
  flex: 0 1 500px;
  padding: 50px;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  border-radius: 5px;
}

.c5:hover {
  background-color: #c32865;
}

.c8 {
  padding: 20px 0;
}

.c7 + .c7 {
  border-top: 1px solid #FFFFFF;
}

.c12 {
  background-color: rgba(255,255,255,0.2);
  -webkit-flex: 0 1 500px;
  -ms-flex: 0 1 500px;
  flex: 0 1 500px;
  padding: 50px;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  border-radius: 5px;
}

.c12:hover {
  background-color: #c32865;
}

.c14 {
  padding: 20px 0;
}

.c13 + .c13 {
  border-top: 1px solid #FFFFFF;
}

@media (max-width:768px) {
  .c2 {
    padding: 20px;
  }
}

@media (max-width:768px) {
  .c0 .c3 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-align-items: stretch;
    -webkit-box-align: stretch;
    -ms-flex-align: stretch;
    align-items: stretch;
  }
}

@media (max-width:768px) {
  .c0 .c1 {
    height: auto;
  }
}

@media (max-width:768px) {
  .c5 {
    -webkit-flex: 0 1 100%;
    -ms-flex: 0 1 100%;
    flex: 0 1 100%;
  }
}

@media (max-width:768px) {
  .c12 {
    -webkit-flex: 0 1 auto;
    -ms-flex: 0 1 auto;
    flex: 0 1 auto;
  }
}

<div>
  <section
    class="c0"
    id="ed_exp"
  >
    <div
      class="c1 c2"
    >
      <div
        class="c3 c4"
      >
        <div
          class="c5"
        >
          <h2
            class="c6"
          >
            Educação
          </h2>
          <div
            class="c7 c8"
          >
            <h3
              class="c9"
            >
              FORMAÇÃO ACADÊMICA
            </h3>
            <h4
              class="c10"
            >
              Unicesumar, Maringá-PR / Out/2017 - Ago/2020
            </h4>
            <p
              class="c11"
            >
              Curso Superior de Tecnologia (CST) em Análise de Sistemas de Computação
            </p>
          </div>
          <div
            class="c7 c8"
          >
            <h3
              class="c9"
            >
              LICENÇAS E CERTIFICADOS
            </h3>
            <h4
              class="c10"
            >
              Udemy
            </h4>
            <p
              class="c11"
            >
              Web Design Completo (HTML, CSS e Javascript)
            </p>
          </div>
        </div>
        <div
          class="c12"
        >
          <h2
            class="c6"
          >
            Experiências
          </h2>
          <div
            class="c13 c14"
          >
            <h4
              class="c10"
            >
              Vip Sistemas, Maringá-PR / Jul/2018 - Dez/2019
            </h4>
            <p
              class="c11"
            >
              Realizava a função de suporte de sistemas de TI, onde trabalhava com a manutenção de servidores (Locais e em nuvem) e banco de dados.
            </p>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
